indico/modules/attachments/templates/_attachments.html

```diff
@@ -8,10 +8,11 @@
 
 {% macro render_attachment(attachment, has_label=false, classes='') %}
     {% set previewable = (attachment.file.is_previewable and not g.static_site) or false %}
+    {% set visible_link = attachment.type.name == 'link' and (not attachment.is_self_protected or attachment.can_access(session.user)) %}
     <a class="attachment {{ get_attachment_icon(attachment) }} {% if attachment.is_self_protected %}is-protected{% endif %} {% if has_label %}has-label{% endif %} {%- if previewable %} js-preview-dialog{% endif %} {{ classes }}"
        data-previewable="{{ previewable|tojson }}"
        data-attachment-id="{{ attachment.id }}"
-       href="{{ attachment.download_url }}" target="_blank"
+       href="{{ attachment.link_url if visible_link else attachment.download_url }}" target="_blank"
```

Comment:
Maybe this would be a good time to start adding `rel="noopener noreferrer"` to those links